modules/common/job/job.go

```diff
@@ -39,15 +39,15 @@ func NewJob(
 	job *batchv1.Job,
 	jobType string,
 	preserve bool,
-	timeout time.Duration,
+	timeout time.Duration, // unused
```

Comment:
wondering if we should still support this, like when 0 passed it is not used and something higher to consider requeue ?

Reply:
My goal is to get rid of the requeue within this helper. I can even drop the field from the Job struct as I anyhow need to do adaptation PRs in other operators so I can do the signature change as well at the same time.

If we return the status of the job to the caller then the caller can decide to requeue or just return. Still I believe that we should never need to explicitly requeue due to a status of a k8s managed resource as we can watch that instead. (The situation can be different for non k8s resource statuses like when we depend on something to happen with openstack or galera itself)

Reply:
should we proceed with this? looks like a valid and nice to have fix

Reply:
Yes, we should. If you have time feel free to pick it up. This needs adaptation in all our service-operators.